packages/fxa-settings/src/models/integrations/base-integration.ts

```diff
@@ -0,0 +1,89 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { SyncIntegrationFeatures } from './base-sync-integration';
+
+// import { ModelDataProvider } from '../../lib/model-data';
+
+// TODO: account for 'pairing' auth-brokers
+export enum IntegrationType {
+  Base,
+  OAuthRedirect,
+  OAuthWebChannel,
+  Sync,
+  SyncChannel,
+  SyncWebChannel,
+  V3Desktop,
+  Web,
+}
+
+export abstract class Integration {
+  type: IntegrationType;
+  abstract features: Partial<IntegrationFeatures | SyncIntegrationFeatures>;
```

Comment:
Btw for any early reviewers, I'm still tweaking all of these classes so feel free to throw suggestions out but it's also WIP and you might see some changes later today, e.g. pretty sure I'm going to separate this out so there's no union type here.